src/app/service/timer-service.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+
+export type Timer = {
+  label: string | null;
+  hours: number | null;
+  minutes: number | null;
+  seconds: number | null;
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TimerService {
+  private timers: Timer[] = [];
+
+  constructor() {}
+
+  addTimer(timer: Timer) {
+    this.timers.push(timer);
+  }
+
+  getTimers(): Observable<Timer[]> {
+    return of(this.timers);
```

Comment:
I would keep it in a stream - not just converting [] into the stream. So I would keep it in Behaviour Subject, then push (next(timer)) and use the scan to do the reducer.

Reply:
Changed to behaviorSubject